collectors/jiraffe/tests/test_collectors.py

```diff
@@ -374,17 +374,32 @@ def test_collect_tracker_with_multi_cve_flaw(self):
 
 
 class TestMetadataCollector:
-    @freeze_time(timezone.datetime(2015, 12, 12))
+    # When re-recording VCRs, for vulntype_exists=False, copy the cassettes recorded for
+    # vulntype_exists=True and manually tweak the cassette file to simulate the Vulnerability
+    # type not existing. This visibly highlights the behavior of MetadataCollector and makes
+    # it visible if there's a bug where all fields of type 1 are discarded when type 12207
+    # is collected (an actual bug experienced during development).
+    # If it works correctly, fields_count with vulntype_exists=True must not be lower than with
+    # vulntype_exists=False.
+    @freeze_time(timezone.datetime(2024, 8, 8))
     @pytest.mark.vcr
-    @pytest.mark.parametrize("project_key,fields_count", [("RHEL", 120), ("OSIM", 20)])
-    def test_collect(self, pin_envs, project_key, fields_count):
+    @pytest.mark.parametrize(
+        "project_key,fields_count,vulntype_exists",
+        [
+            ("RHEL", 125, False),
+            ("OSIM", 18, False),
+            ("RHEL", 135, True),
+            ("OSIM", 19, True),
+        ],
+    )
+    def test_collect(self, pin_envs, project_key, fields_count, vulntype_exists):
```

Comment:
`vulntype_exists` is a preparation for future work? I don't see it used anywhere.

Reply:
The value is not used anywhere, but is used to have a second set of test cassettes, to test the different behavior:

```
    # When re-recording VCRs, for vulntype_exists=False, copy the cassettes recorded for
    # vulntype_exists=True and manually tweak the cassette file to simulate the Vulnerability
    # type not existing. This visibly highlights the behavior of MetadataCollector and makes
    # it visible if there's a bug where all fields of type 1 are discarded when type 12207
    # is collected (an actual bug experienced during development).
    # If it works correctly, fields_count with vulntype_exists=True must not be lower than with
    # vulntype_exists=False.
```

Do you have a suggestion how to make it more clear?

Reply:
Updated the comment to be clearer.

Reply:
I see. Probably I don't have any better solution, keep this, I think it is clear enough.